examples/eady_turbulence.jl

```diff
@@ -5,306 +5,398 @@
+max_Δt = min(grid.Δx / Ū, grid.Δx^4 / κ₄h, grid.Δz^2 / κ₂z)
 
-ζmax = FieldMaximum(abs, ζ)
-δmax = FieldMaximum(abs, δ)
+wizard = TimeStepWizard(cfl=1.0, Δt=0.1*max_Δt, max_change=1.1, max_Δt=max_Δt)
 
-# # Plot setup
+# ## A progress messenger
 #
-# Set `makeplot = true` to make plots as the model runs.
+# We write a function that prints out a helpful progress message while the simulation runs.
 
-makeplot = false
+using Printf
+using Oceananigans.Diagnostics: AdvectiveCFL
 
-using PyPlot, PyCall
+CFL = AdvectiveCFL(wizard)
 
-GridSpec = pyimport("matplotlib.gridspec").GridSpec
+start_time = time_ns()
 
-fig = figure(figsize=(12, 8))
+progress(sim) = @printf("i: % 6d, sim time: % 10s, wall time: % 10s, Δt: % 10s, CFL: %.2e\n",
+                        sim.model.clock.iteration,
+                        prettytime(sim.model.clock.time),
+                        prettytime(1e-9 * (time_ns() - start_time)),
+                        prettytime(sim.Δt.Δt),
+                        CFL(sim.model))
 
-gs = GridSpec(2, 2, height_ratios=[2, 1])
+# ## Build the simulation
+#
+# We're ready to build and run the simulation. We ask for a progress message and time-step update
+# every 100 iterations,
 
-axs = ntuple(4) do i
-    fig.add_subplot(get(gs, i-1))
-end
+simulation = Simulation(model, Δt = wizard, iteration_interval = 10,
+                                                     stop_time = 10day,
+                                                      progress = progress)
 
-function makeplot!(axs, model)
-    nx, ny, nz = size(model.grid)
+# ## Output and plotting
+#
+# To visualize the baroclinic turbulence ensuing in the Eady problem,
+# we use `ComputedField`s to diagnose and output vertical vorticity and divergence.
+# Note that `ComputedField`s take "AbstractOperations" on `Field`s as input:
 
-    xC_xy = repeat(reshape(model.grid.xC, nx, 1), 1, ny)
-    xF_xy = repeat(reshape(model.grid.xF[1:end-1], nx, 1), 1, ny)
+using Oceananigans.AbstractOperations
+using Oceananigans.Fields: ComputedField
 
-    yC_xy = repeat(reshape(model.grid.yC, 1, ny), nx, 1)
-    yF_xy = repeat(reshape(model.grid.yF[1:end-1], 1, ny), nx, 1)
+u, v, w = model.velocities # unpack velocity `Field`s
 
-    xC_xz = repeat(reshape(model.grid.xC, nx, 1), 1, nz+1)
-    xF_xz = repeat(reshape(model.grid.xF[1:end-1], nx, 1), 1, nz)
+## Vertical vorticity [s⁻¹]
+ζ = ComputedField(∂x(v) - ∂y(u))
 
-    zC_xz = repeat(reshape(model.grid.zC, 1, nz),   nx, 1)
-    zF_xz = repeat(reshape(model.grid.zF, 1, nz+1), nx, 1)
+## Horizontal divergence, or ∂x(u) + ∂y(v) [s⁻¹]
+δ = ComputedField(-∂z(w))
 
-    compute!(vertical_vorticity)
-    compute!(divergence)
+# With the vertical vorticity, `ζ`, and the horizontal divergence, `δ` in hand,
+# we create a JLD2OutputWriter that saves `ζ` and `δ` and add it to 
+# `simulation`.
 
-    @printf("\nmax ζ/f: %.2e, max δ/f: %.2e\n\n", ζmax()/f, δmax()/f)
+using Oceananigans.OutputWriters: JLD2OutputWriter
+using Oceananigans.Utils: hour
```

Comment:
There are a few of these scattered throughout the example. Would a single
```julia
using Oceananigans.Utils: second, minute, hour, day
```
at the top of the file be better?

Reply:
I moved this to the top of the file. We only use `hour` and `day`, so I only import those names.